src/main/kotlin/no/nav/syfo/varsel/arbeidstaker/ArbeidstakerVarselApi.kt

```diff
@@ -50,41 +50,17 @@ fun Route.registerArbeidstakerVarselApi(
                 val requestPersonIdent = call.personIdent()
                     ?: throw IllegalArgumentException("No PersonIdent found in token")
 
-                val varselUuid = UUID.fromString(call.parameters[arbeidstakerVarselApiVarselParam])
+                val brevUuid = UUID.fromString(call.parameters[arbeidstakerVarselApiVarselParam])
 
-                val motedeltakerArbeidstakerVarsel = dialogmotedeltakerService.getDialogmoteDeltakerArbeidstakerVarselList(
-                    uuid = varselUuid
-                ).firstOrNull()
-
-                val referat = dialogmoteService.getReferat(
-                    referatUUID = varselUuid
-                )
-
-                if (motedeltakerArbeidstakerVarsel == null && referat == null) {
-                    throw IllegalArgumentException("No Varsel found for PersonIdent with uuid=$varselUuid")
-                }
-                val motedeltakerArbeidstakerId = if (referat != null) {
-                    dialogmoteService.getDialogmote(referat.moteId).arbeidstaker.id
-                } else {
-                    motedeltakerArbeidstakerVarsel!!.motedeltakerArbeidstakerId
-                }
+                val brev = dialogmoteService.getArbeidstakerBrevFromUuid(brevUuid)
```

Comment:
Ser at vi nå må hente referatet fra databasen flere ganger (inni kallet her og for å sjekke om det er referat i updateArbeidstakerBrevSettSomLest), men det er kanskje ikke noe problem? (Og vanskelig å unngå uten større endringer/refaktorere tilbake sånn det var 😅 )

Reply:
Ja, det blir en ekstra lesing av den raden. Men det er et oppslag på en indeksert kolonne, så det har neppe så stor betydning.